lib/sprockets/loader.rb

```diff
@@ -186,7 +186,7 @@ def load_from_unloaded(unloaded)
           dependencies_digest: DigestUtils.digest(resolve_dependencies(metadata[:dependencies]))
         }
 
-        asset[:id]  = pack_hexdigest(digest(asset))
```

Comment:
Asset#hexdigest is only a digest of `source` as opposed to `Asset#id` which is a digest of the "asset" hash.